src/processor/intents/creeps/tick.js

```diff
@@ -94,12 +94,35 @@ module.exports = function(object, roomObjects, roomTerrain, bulk, bulkUsers, roo
         bulk.update(object._id, {fatigue: object.fatigue});
     }
 
-    if(_.isNaN(object.hits)) {
+    if(_.isNaN(object.hits) || object.hits <= 0) {
         require('./_die')(object, roomObjects, bulk, stats);
     }
 
     if(object.userSummoned && _.any(roomObjects, i => i.type == 'creep' && i.user != '2' && i.user != roomController.user)) {
         require('./_die')(object, roomObjects, bulk, stats);
     }
 
+    if (object._recalcBody) {
+        object._recalcBody = undefined;
+
+        if (object.hits > object.hitsMax) {
+            object.hits = object.hitsMax;
+
+            bulk.update(object, {
+                hits: object.hits,
+            });
```

Comment:
OK, that is probably a good idea! In the local driver library, bulk.* calls seem to not actually make any database calls until the end of the tick, and instead just merge different updates together, but that could definitely be different for the official server.